### ISSUE TITLE
Custom Path and some other options not settable

### ISSUE THREAD

**Issue by samyouel** [thathoff/kirby-git-content#103]
Hey!
I'm on a multipage setup, where I define the root paths for the content folders etc in the `/index.php` file like so:
```php
  $kirby = new Kirby([
    'roots' => [
      ...
      'content'  => $dynamic_directory . '/content',
      ...
    ],
  ]);
```

Now I set the content folder path in the settings (in the ready part of the settings, so the dynamic kirby root for content is available) : 
```php
'ready' => function ($kirby) {
    return [
    'thathoff.git-content.path' => $kirby->root("content"),
    ]},
```

### Now the Problem

The settings are completely ignored by the plugin, in the same way the settings for `commitMessage` and `menuLabel` (maybe even more that's the ones I checked). Some of the settings, like `commit`, or `pull` are taken from the settings.

At first I thought it might be because of the `ready` part in my settings but even without that the custom path is not recognised.

In the constructor in `site/plugins/git-content/src/KirbyGitHelper.php:24~29` it says:
```php
    public function __construct($repoPath = false)
    {
        $this->kirby = kirby();
        $this->repoPath = $repoPath ? $repoPath : option('thathoff.git-content.path', $this->kirby->root("content"));
        $this->commitMessageTemplate = option('thathoff.git-content.commitMessage', ':action:(:item:): :url:');
    }
```

It seems like this is fired too early in the chain, even before the settings from the `config.php` are read.

-> Can you reproduce?

I would appreciate it if this works :)
Thank you!

**Comment by thathoff** [comment]
Yes setting the options in the `ready` hook is not supported as the plugin already initialises the KirbyGit class when being loaded. But, the plugin should use the content folder automatically. And it did in all cases I’ve used a different / dynamic content folder. So from my point of view the problem is something different. 🤔

**Comment by samyouel** [comment]
Okay, so the `ready` won't be able to work. Cool, at least that makes sense.

I've tried this again and it's actually still not working. In the meantime I've altered the plugin's source file, but that's obviously not a solution.
Would be cool to find a solution here. Thank you!

**Comment by samyouel** [comment]
Still nothing on this? The config settings for `thathoff.git-content.path` are not available in the `__construct` of `/src/KirbyGitHelper.php`, at that point the config settings aren't loaded yet, even if not defined in the ready state. The setting is completely ignored... it's a shame I have to remember altering the source code every time I do an update of the plugin.

**Comment by thathoff** [comment]
I still think the problem is something else. For testing purposes, I used the [Kirby Compose Kit](https://github.com/getkirby/composerkit) (where the content folder does not use the default path), and the plugin automatically detected the correct path, as we use the default of `$kirby->root('content')` everywhere. Perhaps you set the `thathoff.git-content.path` option in the `config.php` file before, or maybe the Kirby content path is changed later on.

For this reason, I don't think it's worth investing time supporting the change to the `thathoff.git-content.path` configuration option within the 'ready' hook. However, a merge request incorporating this change would be very welcome. 😄 

One last thing: you can use this plugin free of charge, and I have always accepted merge requests. Words like ”shame” feel a bit rude in this context.

**Comment by samyouel** [comment]
Hey, tut mir leid, ich wollte nicht unhöflich oder unfair rüberkommen... (ich schreibe mal deutsch, weil ich glaube dass die Sprache zusätzliche Distanz bringt, hoffe das ist okay für dich für diese Nachricht) das sollte mehr soviel heißen wie, dass es einfach ein wenig schade ist, dass es irgendwie bei mir nicht funktioniert.. Sorry.
Du hast absolut recht, ich hatte es vorher nicht nochmal mit einem Starterkit probiert und bis davon ausgegangen, wenn es in meinem Projekt nicht funktioniert, wird das so auch in anderen Projekten sein. Das war natürlich etwas kurz gedacht.

Ich kann nicht ganz verstehen, warum es in einem anderen Projekt geht, und bei mir nicht. Die Kirbyversion ist die gleiche. der einzige Unterschied ist, dass ich in meiner `index.php` einen custom Pfad für das config file angelegt habe. aber auch das führt in dem Composerkit nicht zu dem gleichen Problem... Bei mir werden die `kirby()->roots()` auch in diesem context einfach gar nicht erkannt, und deswegen gibt der Kirby `option()` helper scheinbar nichts zurück weil er die richtige config root auch gar nicht kennt. Viel weiter komme ich da irgendwie auch nicht. 
...falls du einen weiteren Tipp für mich hast, wäre ich dir sehr dankbar, ich bin auf jeden Fall froh dass es nur ein Fehler auf meiner Seite ist und nichts generelles. Danke dir!

**Comment by thathoff** [comment]
Kein Problem! 😄 Vielleicht checkst du noch mal die Doku zum [Custom folder setup](https://getkirby.com/docs/guide/configuration/custom-folder-setup) bei Kirby, vielleicht ist da irgendwas komisch.

**Comment by thathoff** [comment]
If this issue persists, please reopen it. I'm closing it since it's been inactive for a couple of months.
